File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/TeamStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class TeamStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "teamName",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private TeamStateEventId value;

    public TeamStateEventIdFlattenedDto()
    {
        this(new TeamStateEventId());
    }

    public TeamStateEventIdFlattenedDto(TeamStateEventId value)
    {
        this.value = value;
    }

    public TeamStateEventId toTeamStateEventId()
    {
        return this.value;
    }


    public String getTeamName()
    {
        return this.value.getTeamName();
    }

    public void setTeamName(String teamName)
    {
        this.value.setTeamName(teamName);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        TeamStateEventIdFlattenedDto other = (TeamStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/TeamStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class TeamStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'teamName' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var TeamStateEventId
     */
    private $value;

    /**
     * @param TeamStateEventId $value
     */
    public function __construct(TeamStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new TeamStateEventId();
        }
    }

    /**
     * @return TeamStateEventId
     */
    public function toTeamStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getTeamName()
    {
        return $this->value->getTeamName();
    }

    /**
     * @param string $teamName
     */
    public function setTeamName($teamName)
    {
        $this->value->setTeamName($teamName);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getTeamName(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return TeamStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setTeamName($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

